src/main/resources/site/parts/banner/banner.ts

```diff
@@ -0,0 +1,126 @@
+import {Request, Response} from "enonic-types/controller";
+import {ResourceKey} from "enonic-types/thymeleaf";
+import {Content} from "enonic-types/content";
+import {Component} from 'enonic-types/portal'
+import {BannerPartConfig} from "./banner-part-config";
+import {MunicipalityWithCounty} from "../../../lib/ssb/dataset/klass/municipalities";
+
+const {
+  getContent,
+  getComponent,
+  imageUrl
+} = __non_webpack_require__('/lib/xp/portal')
+const {
+  render
+} = __non_webpack_require__('/lib/thymeleaf')
+const {
+  getMunicipality, removeCountyFromMunicipalityName
+} = __non_webpack_require__('/lib/ssb/dataset/klass/municipalities')
+const {
+  getImageAlt
+} = __non_webpack_require__('/lib/ssb/utils/imageUtils')
+const {
+  renderError
+} = __non_webpack_require__('/lib/ssb/error/error')
+
+const i18nLib = __non_webpack_require__('/lib/xp/i18n')
+const view: ResourceKey = resolve('./banner.html')
+
+exports.get = function (req: Request) {
+  try {
+    return renderPart(req)
+  } catch (e) {
+    return renderError(req, 'Error in part', e)
+  }
+}
+
+exports.preview = (req: Request): Response => renderPart(req)
+
+function renderPart(req: Request): Response {
+  const page: Content<any> = getContent() // TODO finn content type her
+  const part: Component<BannerPartConfig> = getComponent()
+  const pageType: PageTypeGeneral | PageTypeFaktaside | PageTypeKommunefakta = part.config.pageType
```

Comment:
Suggestion: PageTypeGeneral | PageTypeFaktaside | PageTypeKommunefakta --> BannerPartConfig['pageType'] ? This way, the types in the automatically generated part config can be used and we won't have to define an interface manually.